File: app/src/main/java/co/early/password123/feature/networkmonitor/DoubleCheckConnection.java
package co.early.password123.feature.networkmonitor;

import java.io.IOException;

import co.early.asaf.core.logging.AndroidLogger;
import co.early.asaf.retrofit.InterceptorLogging;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;


public class DoubleCheckConnection {

    private final static String TAG = DoubleCheckConnection.class.getSimpleName();


    public boolean canWeConnect() {

        OkHttpClient client = new OkHttpClient().newBuilder()
                .addInterceptor(new InterceptorLogging(new AndroidLogger())).build();

        Request request = new Request.Builder().url("https://google.com").head().build();
        Response response = null;
        try {
            response = client.newCall(request).execute();
            return response.isSuccessful();
        } catch (IOException e) {
            return false;
        } finally {
            if (response != null) {
                response.body().close();
            }
        }
    }

}

File: app/src/main/java/co/early/password123/feature/networkmonitor/DoubleCheckConnection.kt
package co.early.password123.feature.networkmonitor

import co.early.asaf.core.logging.AndroidLogger
import co.early.asaf.core.logging.Logger
import co.early.asaf.retrofit.InterceptorLogging
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import java.io.IOException


class DoubleCheckConnection {

    @JvmOverloads
    fun canWeConnect(logger: Logger = AndroidLogger()): Boolean {

        val client = OkHttpClient().newBuilder()
                .addInterceptor(InterceptorLogging(logger)).build()

        val request = Request.Builder().url("https://google.com").head().build()
        var response: Response? = null
        try {
            response = client.newCall(request).execute()
            return response!!.isSuccessful
        } catch (e: IOException) {
            return false
        } finally {
            if (response != null) {
                response.body()!!.close()
            }
        }
    }

    companion object {
        private val TAG = DoubleCheckConnection::class.java.simpleName
    }

}
